cli/src/commands/git/clone.rs

```diff
@@ -227,11 +232,30 @@ fn do_git_clone(
         )
     })
     .map_err(|err| match err {
-        GitFetchError::NoSuchRemote(_) => {
-            panic!("shouldn't happen as we just created the git remote")
+        GitFetchError::NoSuchRemote(repo_name) => {
+            user_error(format!("Could not find repository at '{repo_name}'"))
+        }
+        GitFetchError::GitNotFound(path) => {
+            if let Some(git_path) = path {
+                user_error(format!("Could not execute provided git path: `{git_path}`"))
+            } else {
+                user_error("Could not find a `git` executable in the OS path")
+            }
         }
         GitFetchError::GitImportError(err) => CommandError::from(err),
         GitFetchError::InternalGitError(err) => map_git_error(err),
+        GitFetchError::GitForkError(err) => CommandError::with_message(
+            crate::command_error::CommandErrorKind::Internal,
+            "External git process failed",
+            err,
+        ),
+        GitFetchError::GitExternalError(err) => {
+            CommandError::new(crate::command_error::CommandErrorKind::Internal, err)
+        }
+        GitFetchError::PathConversionError(path) => CommandError::new(
+            crate::command_error::CommandErrorKind::Internal,
+            format!("Failed to convert path {} to string", path.display()),
+        ),
```

Comment:
nits:
- there are `internal_error`/`_with_message()` helper functions
- might be better to extract uninteresting error variants as `GitFetchError::Subprocess(GitSubprocessError)` or `GitFetchError::InternalGit(Box<dyn ...>)`